src/components/PlayerWelcome.js

```diff
@@ -1,75 +1,116 @@
+import React, { useState, useEffect } from "react";
 import { useHistory } from 'react-router-dom'
 import { Button } from "react-bootstrap";
+import { getFirestore, getDoc, doc} from 'firebase/firestore'
 import SignUp1 from "../assets/images/PlayerWelcome/SignUp1.png"
 import SignUp2 from "../assets/images/PlayerWelcome/SignUp2.png"
 import Vote from "../assets/images/PlayerWelcome/Vote1.png"
-import Calendar from "../assets/images/PlayerWelcome/Complete1.svg"
-import Top from '../assets/images/PlayerWelcome/Top.svg'
+import BlackCurve from '../assets/images/PlayerWelcome/BlackCurve.svg'
 import './PlayerWelcome.scss'
+import { auth } from "../firebase";
+import { dummyPassword } from "../constants";
+import { useAuth } from "../contexts/AuthContext";
 
 export default function PlayerWelcome() {
     const history = useHistory(); 
+    const { currentUser } = useAuth();
+    const [ loading, setLoading ] = useState(true);
+    const [ challengerInfo, setChallengerInfo ] = useState(null);
 
+    const url = new URL(window.location.href)
+    const code = url.searchParams.get("code")
+
+    async function getChallengerInfo() {
```

Comment:
Either in this function or wherever fit, can we save the challenger's code to firebase after getting it?

The code needs to be updated in the player's (current user's) document

Here is a player's document right now:
![image](https://user-images.githubusercontent.com/66650721/159108445-686e29a9-350c-47fa-9259-329f0b831146.png)

We want to update the `invitedBy` string to include the challenger's code; this will allow us to update the challenger's badges when the player completes an action

Reply:
What I have now successfully updates `invitedBy` after the player signs in, but not signs up. Could you take a look at this? For your reference, the function `addInvitedBy` is called on each of the action pages.

Reply:
This is resolved in our conversation right? 

Does my comment on this make sense?: https://github.com/8by8-org/web-app/issues/93#issuecomment-1083987691 ?

Reply:
Yes, I currently have what you've described for the alpha release. My only problem was getting the challenger info to save to firebase after the player signed up and landed on the action page.